File: src/main/java/com/cdac/flowflix/service/WatchlistService.java
package com.cdac.flowflix.service;

import java.util.List;

import com.cdac.flowflix.dto.WatchlistDTO;

public interface WatchlistService {

    // ==========================================
    // ADD MOVIE
    // ==========================================

    String addToWatchlist(Long movieId);

    // ==========================================
    // REMOVE MOVIE
    // ==========================================

    String removeFromWatchlist(Long movieId);

    // ==========================================
    // GET MY WATCHLIST
    // ==========================================

    List<WatchlistDTO> getMyWatchlist();

    // ==========================================
    // CHECK MOVIE EXISTS
    // ==========================================

    boolean existsInWatchlist(Long movieId);

    // ==========================================
    // CLEAR WATCHLIST
    // ==========================================

    String clearWatchlist();

    // ==========================================
    // WATCHLIST COUNT
    // ==========================================

    Long getWatchlistCount();

}

File: frontend/src/services/watchlistService.js
import api from './api';

// GET    /watchlist              → get logged-in user's watchlist
const getWatchlist = () => api.get('/watchlist');

// POST   /watchlist/:movieId     → add movie to watchlist
const addToWatchlist = (movieId) => api.post(`/watchlist/${movieId}`);

// DELETE /watchlist/:movieId     → remove from watchlist
const removeFromWatchlist = (movieId) => api.delete(`/watchlist/${movieId}`);

// GET    /watchlist/check/:movieId → check if movie is in watchlist
const isInWatchlist = (movieId) => api.get(`/watchlist/check/${movieId}`);

const watchlistService = { getWatchlist, addToWatchlist, removeFromWatchlist, isInWatchlist };
export default watchlistService;
